api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/SchoolService.java

```diff
@@ -64,55 +73,117 @@ public void loadSchoolsIntoRedisCache(List<ca.bc.gov.educ.api.trax.model.dto.ins
 				.saveAll(schoolTransformer.transformToEntity(schools));
 	}
 
-    /*public SchoolDetail getCommonSchoolDetailById(String schoolId, String accessToken) {
+	public List<School> getSchoolsFromRedisCache() {
+		log.debug("**** Getting schools from Redis Cache.");
+		return  schoolTransformer.transformToDTO(schoolRedisRepository.findAll());
+	}
+
+	public void initializeSchoolCache(boolean force) {
+		String cacheStatus = redisTemplate.opsForValue().get(CacheKey.SCHOOL_CACHE.name());
+		cacheStatus = cacheStatus == null ? "" : cacheStatus;
+		if (CacheStatus.LOADING.name().compareToIgnoreCase(cacheStatus) == 0
+				|| CacheStatus.READY.name().compareToIgnoreCase(cacheStatus) == 0) {
+			log.info(String.format("SCHOOL_CACHE status: %s", cacheStatus));
+			if (force) {
+				log.info("Force Flag is true. Reloading SCHOOL_CACHE...");
+				redisTemplate.opsForValue().set(CacheKey.SCHOOL_CACHE.name(), CacheStatus.LOADING.name());
+				loadSchoolsIntoRedisCache(getSchoolsFromInstituteApi());
+				redisTemplate.opsForValue().set(CacheKey.SCHOOL_CACHE.name(), CacheStatus.READY.name());
+				log.info("SUCCESS! - SCHOOL_CACHE is now READY");
+			} else {
+				log.info("Force Flag is false. Skipping SCHOOL_CACHE reload");
+			}
+		} else {
+			log.info("Loading SCHOOL_CACHE...");
+			redisTemplate.opsForValue().set(CacheKey.SCHOOL_CACHE.name(), CacheStatus.LOADING.name());
+			loadSchoolsIntoRedisCache(getSchoolsFromInstituteApi());
+			redisTemplate.opsForValue().set(CacheKey.SCHOOL_CACHE.name(), CacheStatus.READY.name());
+			log.info("SUCCESS! - SCHOOL_CACHE is now READY");
+		}
+	}
+
+	public SchoolDetail getSchoolDetailByIdFromInstituteApi(String schoolId) {
+		return getSchoolDetailByIdFromInstituteApi(schoolId, "");
+	}
+
+    public SchoolDetail getSchoolDetailByIdFromInstituteApi(String schoolId, String accessToken) {
         try {
             return webClient.get().uri(
                             String.format(constants.getSchoolDetailsByIdFromInstituteApiUrl(), schoolId))
                     .headers(h -> {
```

Comment:
Create a webclient and qualifier for grad-edx-client like the pattern established [here](https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/main/api/src/main/java/ca/bc/gov/educ/api/trax/config/GradTraxConfig.java#L56) and used [here](https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/main/api/src/main/java/ca/bc/gov/educ/api/trax/util/CommonSchoolCache.java#L29) and configured [here](https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/main/api/src/main/resources/application.yaml#L47). Also we should add some resilience to the call in case of a 5xx response (like 503 due to pod restarts, etc.) There is an example of how you can add simple resilience to REST calls [here](https://github.com/bcgov/EDUC-GRAD-GRADUATION-API/blob/main/api/src/main/java/ca/bc/gov/educ/api/graduation/service/RESTService.java#L73)

Reply:
WebClient Fixed:
https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/c79bdcc841ed4529314fd9ef0d3bd821fb053b15/api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/SchoolService.java#L31

Reply:
Resilience Added:
https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/c79bdcc841ed4529314fd9ef0d3bd821fb053b15/api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/SchoolService.java#L78